parley/src/context.rs

```diff
@@ -159,20 +171,36 @@ impl<'a, B: Brush, T: TextSource> RangedBuilder<'a, B, T> {
                 line_height: s.line_height,
             }
         }));
+
+        // Sort the inline boxes as subsequent code assumes that they are in text index order.
+        // Note: It's important that this is a stable sort to allow users to control the order of contiguous inline boxes
+        lcx.inline_boxes.sort_by_key(|b| b.index);
```

Comment:
For better or worse, parley was intentionally designed to avoid transient heap allocation and a non-trivial chunk of the complexity is a result of that. So I’d like to avoid letting “allocation per item” collections creep in and break that property. There are already a couple cases of that in this PR (including the stable sort here) but I believe they will all be corrected in future work.

Reply:
I mean, it doesn't have to be `BTreeMap`, but I guess that's one more dependency.